extensions/azdata/src/azdata.ts

```diff
@@ -0,0 +1,115 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as os from 'os';
+import * as path from 'path';
+import * as uuid from 'uuid';
+import * as which from 'which';
+import * as vscode from 'vscode';
+import { HttpClient } from './common/httpClient';
+import * as loc from './localizedConstants';
+import { executeCommand } from './common/childProcess';
+
+/**
+ * Information about an azdata installation
+ */
+export interface IAzdata {
+	path: string,
+	version: string
+}
+
+/**
+ * Finds the existing installation of azdata, or throws an error if it couldn't find it
+ * or encountered an unexpected error.
+ * @param outputChannel Channel used to display diagnostic information
+ */
+export async function findAzdata(outputChannel: vscode.OutputChannel): Promise<IAzdata> {
+	outputChannel.appendLine(loc.searchingForAzdata);
+	try {
+		let azdata: IAzdata | undefined = undefined;
+		switch (process.platform) {
+			case 'darwin':
+				azdata = await findAzdataDarwin(outputChannel);
+				break;
+			case 'win32':
+				azdata = await findAzdataWin32(outputChannel);
+				break;
+			default:
+				azdata = await findSpecificAzdata('azdata', outputChannel);
+		}
+		outputChannel.appendLine(loc.foundExistingAzdata(azdata.path, azdata.version));
+		return azdata;
+	} catch (err) {
+		outputChannel.appendLine(loc.couldNotFindAzdata(err));
+		throw err;
+	}
+}
+
+/**
+ * Downloads the appropriate installer and/or runs the command to install azdata
+ * @param outputChannel Channel used to display diagnostic information
+ */
+export async function downloadAndInstallAzdata(outputChannel: vscode.OutputChannel): Promise<void> {
+	const statusDisposable = vscode.window.setStatusBarMessage(loc.installingAzdata);
+	try {
+		switch (process.platform) {
+			case 'win32': await downloadAndInstallAzdataWin32(outputChannel);
+		}
+	} finally {
+		statusDisposable.dispose();
+	}
+}
+
+/**
+ * Downloads the Windows installer and runs it
+ * @param outputChannel Channel used to display diagnostic information
+ */
+async function downloadAndInstallAzdataWin32(outputChannel: vscode.OutputChannel): Promise<void> {
+	const downloadPath = path.join(os.tmpdir(), `azdata-msi-${uuid.v4()}.msi`);
```

Comment:
I've noticed that in some extensions we used our own implementation of GUIDs, should we use this instead?

Reply:
I'd say it's a bit safer from a code point of view. But adding more package dependencies isn't the greatest either. I'll probably actually just move this to use a timestamp instead since it doesn't really need to be unique in this way.